src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -472,6 +494,69 @@ private static String addPrefix(String server) {
         else return "ldap://" + server;
     }
 
+    private String generateId() {
+        return generateId(server, rootDN, userSearchBase, userSearch);
+    }
+
+    @Restricted(NoExternalUse.class)
+    static String generateId(String serverUrl, String rootDN, String userSearchBase, String userSearch) {
+        final MessageDigest digest = DigestUtils.getMd5Digest();
+        digest.update(normalizeServer(serverUrl).getBytes(Charsets.UTF_8));
+        String userSearchBaseNormalized = normalizeUserSearchBase(rootDN, userSearchBase);
+        if (isNotBlank(userSearchBaseNormalized)) {
+            digest.update(userSearchBaseNormalized.getBytes(Charsets.UTF_8));
+        } else {
+            digest.update(new byte[]{0});
+        }
+        if (isNotBlank(userSearch)) {
+            digest.update(userSearch.getBytes(Charsets.UTF_8));
+        } else {
+            digest.update(LDAPConfigurationDescriptor.DEFAULT_USER_SEARCH.getBytes(Charsets.UTF_8));
+        }
+        return Base64.encodeBase64String(digest.digest());
+    }
+
+    private static String normalizeUserSearchBase(String rootDN, String userSearchBase) {
+        if (isBlank(rootDN) && isBlank(userSearchBase)) {
+            return "";
+        }
+        if (isBlank(rootDN)) {
+            return userSearchBase;
+        }
+        if (isBlank(userSearchBase)) {
+            return rootDN;
+        }
+        rootDN = rootDN.trim();
+        userSearchBase = userSearchBase.trim();
+        if (userSearchBase.endsWith(rootDN)) {
```

Comment:
The *actual search* will always be relative to the root DN AIUI so adding it just means the effective search is `userBase + rootDn + rootDn` which is silly